runtime/cudaq/platform/default/rest_server/helpers/RestRemoteServer.cpp

```diff
@@ -549,21 +550,55 @@ class RemoteRestRuntimeServer : public cudaq::RemoteRuntimeServer {
     if (!module)
       throw std::runtime_error("Failed to parse the input MLIR code");
     auto engine = jitMlirCode(*module, passes);
+    llvm::SmallVector<void *> returnArg;
     const std::string entryPointFunc =
         std::string(cudaq::runtime::cudaqGenPrefixName) + entryPointFn;
-    auto fnPtr =
-        getValueOrThrow(engine->lookup(entryPointFunc),
-                        "Failed to look up entry-point function symbol");
-    if (!fnPtr)
-      throw std::runtime_error("Failed to get entry function");
+    if (auto funcOp = module->lookupSymbol<LLVM::LLVMFuncOp>(entryPointFunc)) {
+      auto funcTy = funcOp.getFunctionType();
+      auto returnTy = funcTy.getReturnType();
+      // These are the returned types that we support.
+      if (returnTy.isF32()) {
+        io_context.invocationResultBuffer.resize(sizeof(float));
+        returnArg.push_back(io_context.invocationResultBuffer.data());
+      } else if (returnTy.isF64()) {
+        io_context.invocationResultBuffer.resize(sizeof(double));
+        returnArg.push_back(io_context.invocationResultBuffer.data());
+      } else if (returnTy.isInteger(1)) {
+        io_context.invocationResultBuffer.resize(sizeof(bool));
+        returnArg.push_back(io_context.invocationResultBuffer.data());
+      } else if (returnTy.isIntOrIndex()) {
+        io_context.invocationResultBuffer.resize(
+            (returnTy.getIntOrFloatBitWidth() + 7) / 8);
+        returnArg.push_back(io_context.invocationResultBuffer.data());
+      }
+    }
 
-    auto fn = reinterpret_cast<void (*)()>(fnPtr);
-    simulationStart = std::chrono::high_resolution_clock::now();
-    for (std::size_t i = 0; i < numTimes; ++i) {
-      // Invoke the kernel
-      fn();
-      if (postExecCallback) {
-        postExecCallback(i);
+    // Note: currently, we only return data from kernel on single-shot
+    // execution. Once we enable arbitrary sample return type, we can run this
+    // in a loop and return a vector of return type.
+    if (numTimes == 1 && !returnArg.empty()) {
+      simulationStart = std::chrono::high_resolution_clock::now();
+      llvm::Error error = engine->invokePacked(entryPointFunc, returnArg);
+      if (error) {
+        throw std::runtime_error("JIT invocation failed");
+      }
```

Comment:
```suggestion
      if (error) 
        throw std::runtime_error("JIT invocation failed");
```
Consistent style with next statement.